src/contents/button.tsx

```diff
@@ -32,6 +38,29 @@ class ButtonContributionManager {
     }
   }
 
+  /**
+   * Provides a fast check to see if the current URL is on a supported site.
+   */
+  private isSiteSuitable() {
+    const head = document.head;
+    const metaApplication = head.querySelector("meta[name=application-name]");
+    const ogApplication = head.querySelector("meta[property='og:site_name']");
+
+    if (ogApplication && ["GitHub", "GitLab"].includes(ogApplication.getAttribute("content"))) {
```

Comment:
would be great to test this logic so we get informed when one if the sites change and break this.